osu.Game.Rulesets.Catch.Tests/TestSceneHyperDashColouring.cs

```diff
@@ -0,0 +1,306 @@
+        {
+            DrawableFruit drawableFruit = null;
+
+            AddStep("setup hyper-dash fruit", () =>
+            {
+                var fruit = new Fruit { HyperDashTarget = new Banana() };
+                fruit.ApplyDefaults(new ControlPointInfo(), new BeatmapDifficulty());
+
+                Child = setupSkinHierarchy(
+                    drawableFruit = new DrawableFruit(fruit)
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(4f),
+                    }, true, false, false, legacyFruit);
+            });
+
+            AddAssert("hyper-dash fruit colour falls back to catcher colour from skin", () =>
+                legacyFruit
+                    ? checkLegacyFruitHyperDashColour(drawableFruit, TestSkin.CustomHyperDashColour)
+                    : checkFruitHyperDashColour(drawableFruit, TestSkin.CustomHyperDashColour));
+        }
+
+        private Drawable setupSkinHierarchy(Drawable child, bool customCatcherColour, bool customAfterColour, bool customFruitColour, bool legacySkin = true)
+        {
+            var testSkinProvider = new SkinProvidingContainer(new TestSkin(customCatcherColour, customAfterColour, customFruitColour));
+
+            if (legacySkin)
+            {
+                var legacySkinProvider = new SkinProvidingContainer(skins.GetSkin(DefaultLegacySkin.Info));
+                var legacySkinTransformer = new SkinProvidingContainer(new CatchLegacySkinTransformer(testSkinProvider));
+
+                return legacySkinProvider
+                    .WithChild(testSkinProvider
+                        .WithChild(legacySkinTransformer
+                            .WithChild(child)));
+            }
+
+            return testSkinProvider.WithChild(child);
+        }
+
+        private bool checkFruitHyperDashColour(DrawableFruit fruit, Color4 expectedColour) =>
+            fruit.ChildrenOfType<SkinnableDrawable>().First().Drawable.ChildrenOfType<Circle>().Single(c => c.BorderColour == expectedColour).Any(d => d.Colour == expectedColour);
+
+        private bool checkLegacyFruitHyperDashColour(DrawableFruit fruit, Color4 expectedColour) =>
+            fruit.ChildrenOfType<SkinnableDrawable>().First().Drawable.ChildrenOfType<Sprite>().Any(c => c.Colour == expectedColour);
+
+        private class TestSkin : ISkin
+        {
+            public static Color4 CustomHyperDashColour { get; } = Color4.Goldenrod;
+            public static Color4 CustomHyperDashFruitColour { get; } = Color4.Cyan;
+            public static Color4 CustomHyperDashAfterColour { get; } = Color4.Lime;
+
+            private readonly bool customCatcherColour;
+            private readonly bool customAfterColour;
+            private readonly bool customFruitColour;
+
+            public TestSkin(bool customCatcherColour, bool customAfterColour, bool customFruitColour)
```

Comment:
bool bool bool is so hard to read. why not just make the custom color4 properties nullable and settable and set them in the tests? at least the tests would be legible then.